packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts

```diff
@@ -199,6 +199,7 @@ export class Trail extends Resource {
 
   private s3bucket: s3.IBucket;
   private eventSelectors: EventSelector[] = [];
+  private topic: sns.ITopic | undefined;
```

Comment:
```suggestion
  private topic?: sns.ITopic;
```

does this need to be a property on the class? it doesn't seem like it's used outside of the constructor

Reply:
it's a private property, and it makes the bellow `props.topic?.grantPublish(cloudTrailPrincipal)` cleaner, we don't have a strict style in the CDK so I wouldn't insist.

Reply:
i'm not particularly picky here either. I personally find the latter cleaner but let's stick with it as it is because I do think we should maintain consistency within the class at least.